lychee-bin/src/options.rs

```diff
@@ -628,15 +628,48 @@ separated list of accepted status codes. This example will accept 200, 201,
     #[serde(skip)]
     pub(crate) base: Option<Base>,
 
-    /// Base URL used to resolve relative URLs during link checking
+    /// Base URL used to resolve relative URLs in local files.
     /// Example: <https://example.com>
-    #[arg(short, long, value_parser= parse_base)]
+    #[arg(
+        short,
+        long,
+        value_parser = parse_base,
+        long_help = "Base URL to use when resolving relative URLs in local files. If specified,
+relative links in local files are interpreted as being relative to the given
+base URL.
+
+For example, given a base URL of `https://example.com/dir/page`, the link `a`
+would resolve to `https://example.com/dir/a` and the link `/b` would resolve
+to `https://example.com/b`. This behavior is not affected by the filesystem
+path of the file containing these links.
+
+Note that relative URLs without a leading slash become siblings of the base
+URL. If, instead, the base URL ended in a slash, the link would become a child
+of the base URL. For example, a base URL of `https://example.com/dir/page/` and
+a link of `a` would resolve to `https://example.com/dir/page/a`.
+
+Basically, the base URL option resolves links as if the local files were hosted
+at the given base URL address."
+    )]
     #[serde(default)]
     pub(crate) base_url: Option<Base>,
 
-    /// Root path to use when checking absolute local links,
-    /// must be an absolute path
-    #[arg(long)]
+    /// Root path to use when checking absolute links in local files.
```

Comment:
Should we use the opportunity to make this change as well?

```suggestion
    /// Root directory to use when checking absolute links in local files.
```

(Same for `long_help`.)